libs/@local/graph/postgres-store/src/store/postgres/seed_policies.rs

```diff
@@ -0,0 +1,333 @@
+use core::iter;
+
+use error_stack::{Report, ResultExt as _};
+use hash_graph_authorization::{
+    AuthorizationApi,
+    policies::{
+        Effect,
+        action::ActionName,
+        principal::PrincipalConstraint,
+        resource::{
+            EntityResourceConstraint, EntityResourceFilter, EntityTypeResourceConstraint,
+            EntityTypeResourceFilter, ResourceConstraint,
+        },
+        store::{
+            PolicyCreationParams, PolicyFilter, PolicyStore as _, PrincipalFilter,
+            error::EnsureSystemPoliciesError,
+        },
+    },
+};
+use type_system::{
+    ontology::{BaseUrl, VersionedUrl, id::OntologyTypeVersion},
+    principal::{
+        actor::{ActorId, ActorType},
+        role::{RoleId, TeamRole, WebRole},
+    },
+};
+
+use super::{AsClient, PostgresStore};
+
+macro_rules! base_url {
+    ($url:expr) => {
+        BaseUrl::new($url.to_owned()).expect("should be a valid base URL")
+    };
+}
+
+/// Creates a list of filters for entities with their type versions from 1 to `max_version`.
+// TODO: Allow entity filter for only the BaseURL
+//   see https://linear.app/hash/issue/H-4599/support-baseurl-and-version-filter-of-types-in-entity-resource
+fn create_version_filters(
+    base_url: BaseUrl,
+    max_version: u32,
+) -> impl Iterator<Item = EntityResourceFilter> {
+    let mut base_url = Some(base_url);
+
+    (1..=max_version).filter_map(move |version| {
+        Some(EntityResourceFilter::IsOfType {
+            entity_type: VersionedUrl {
+                base_url: if version < max_version {
+                    base_url.as_ref()?.clone()
+                } else {
+                    base_url.take()?
+                },
+                version: OntologyTypeVersion::new(version),
+            },
+        })
+    })
+}
+
+pub(crate) fn system_actor_create_web_policy(
+    system_machine_actor: ActorId,
+) -> PolicyCreationParams {
+    PolicyCreationParams {
+        name: Some("system-machine-create-web".to_owned()),
+        effect: Effect::Permit,
+        principal: Some(PrincipalConstraint::Actor {
+            actor: system_machine_actor,
+        }),
+        actions: vec![ActionName::CreateWeb],
+        resource: None,
+    }
+}
+
+fn system_actor_view_entity_policies(
+    system_machine_actor: ActorId,
+) -> impl Iterator<Item = PolicyCreationParams> {
```

Comment:
why is this an iterator, but the above is not?

Reply:
I expect to add more things here in the future. It's also only called in an iterator context, so I can simply chain them.